components/sync/driver/brave_sync_service_impl.cc

```diff
@@ -305,4 +307,40 @@ void BraveSyncServiceImpl::LocalDeviceAppeared() {
   SyncServiceImpl::NotifyObservers();
 }
 
+namespace {
+const int kCyclesBeforeUpdateP3AObjects = 10;
+}  // namespace
+
+void BraveSyncServiceImpl::OnSyncCycleCompleted(
+    const SyncCycleSnapshot& snapshot) {
+  SyncServiceImpl::OnSyncCycleCompleted(snapshot);
+  if (completed_cycles_count_ % kCyclesBeforeUpdateP3AObjects == 0) {
```

Comment:
nit. I think it might be helpful to have a comment here explaining what is being checked in this conditional, as the modulo there seems a bit elusive to me.